public/components/register_model/model_configuration.tsx

```diff
@@ -3,101 +3,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useState } from 'react';
-import {
-  EuiFormRow,
-  EuiCodeEditor,
-  EuiText,
-  EuiTextColor,
-  EuiCode,
-  EuiSpacer,
-  EuiButtonEmpty,
-} from '@elastic/eui';
-import { useController, useFormContext } from 'react-hook-form';
+import React from 'react';
+import { EuiText, EuiTextColor, EuiCode, EuiSpacer } from '@elastic/eui';
 
 import '../../ace-themes/sql_console.js';
-import { FORM_ITEM_WIDTH } from './form_constants';
-import type { ModelFileFormData, ModelUrlFormData } from './register_model.types';
-import { HelpFlyout } from './help_flyout';
-import { CUSTOM_FORM_ERROR_TYPES } from './constants';
-import { ErrorMessage } from './error_message';
-
-function validateConfigurationObject(value: string) {
-  try {
-    JSON.parse(value.trim());
-  } catch {
-    return 'JSON is invalid. Enter a valid JSON';
-  }
-  return true;
-}
-
-function validateModelType(value: string) {
-  try {
-    const config = JSON.parse(value.trim());
-    if (!('model_type' in config)) {
-      return 'model_type is required. Specify the model_type';
-    }
-  } catch {
-    return true;
-  }
-  return true;
-}
-
-function validateModelTypeValue(value: string) {
-  try {
-    const config = JSON.parse(value.trim());
-    if ('model_type' in config && typeof config.model_type !== 'string') {
-      return 'model_type must be a string';
-    }
-  } catch {
-    return true;
-  }
-  return true;
-}
-
-function validateEmbeddingDimensionValue(value: string) {
-  try {
-    const config = JSON.parse(value.trim());
-    if ('embedding_dimension' in config && typeof config.embedding_dimension !== 'number') {
-      return 'embedding_dimension must be a number';
-    }
-  } catch {
-    return true;
-  }
-
-  return true;
-}
-
-function validateFrameworkTypeValue(value: string) {
-  try {
-    const config = JSON.parse(value.trim());
-    if ('framework_type' in config && typeof config.framework_type !== 'string') {
-      return 'framework_type must be a string';
-    }
-  } catch {
-    return true;
-  }
-  return true;
-}
+import { ModelConfiguration } from '../common/forms/model_configuration';
```

Comment:
Do we need to expose these form components and constants to common folder?

Reply:
You mean components like `ModelConfiguration`?

Reply:
I mean expose `common/forms/model_configuration` contents to `common/forms/index.ts`. Then we can use `import {ModelConfiguration} from '../common'` to import this component. That's not a big issue.